scripts/update_version.py

```diff
@@ -23,12 +23,13 @@
 # Update version string.
 pattern1 = re.compile('assert numpyro.__version__.startswith\\("[^"]*"\\)')
 pattern2 = re.compile("assert numpyro.__version__.startswith\\('[^']*'\\)")
-text = f"assert numpyro.__version__.startswith({new_version})"
+text1 = f"assert numpyro.__version__.startswith({new_version})"
+text2 = text1.replace('"', "'")
 for filename in filenames:
     with open(filename) as f:
         old_text = f.read()
-    new_text = pattern1.sub(text, old_text)
-    new_text = pattern2.sub(text, new_text)
+    new_text = pattern1.sub(text1, old_text)
+    new_text = pattern2.sub(text2, new_text)
```

Comment:
The old pattern changes `'0.6.0'` in notebook to `"0.7.0"` (which needs to be `\"0.7.0\"` to make jupyter understand) but what we want is `'0.7.0'`.

Reply:
You could use [nbQA](https://github.com/nbQA-dev/nbQA) for this:
```
$ nbqa scripts.update_version notebooks/ --nbqa-diff
Cell 2
------
--- notebooks/source/time_series_forecasting.ipynb
+++ notebooks/source/time_series_forecasting.ipynb
@@ -17,4 +17,4 @@
     set_matplotlib_formats("svg")
 
 numpyro.set_host_device_count(4)
-assert numpyro.__version__.startswith("0.5.0")
+assert numpyro.__version__.startswith("0.7.0")

Cell 9
------
--- notebooks/source/discrete_imputation.ipynb
+++ notebooks/source/discrete_imputation.ipynb
@@ -31,4 +31,4 @@
             
         ### outcome model 
         eta_Y = b_A * Aimp + b_B * B
-        sample("obs_Y", dist.Normal(eta_Y, s_Y), obs=Y)

+        sample("obs_Y", dist.Normal(eta_Y, s_Y), obs=Y)

updating /home/marco/numpyro-dev/notebooks/source/time_series_forecasting.ipynb
To apply these changes use `--nbqa-mutate` instead of `--nbqa-diff`
```
The change in cell 9 is just to remove a trailing newline, [which nbQA always does when using modifying notebooks](https://nbqa.readthedocs.io/en/latest/configuration.html#allow-mutations)

---

disclaimer: I'm the author of nbQA

Reply:
Thanks for the nice tool, @MarcoGorelli! I will start using it in my own repositories. 💯